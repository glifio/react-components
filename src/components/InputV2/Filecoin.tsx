import PropTypes from 'prop-types'
import { useEffect, useState } from 'react'
import { FilecoinNumber } from '@glif/filecoin-number'
import { BaseInput, BaseInputProps, BaseInputPropTypes } from './Base'
import { FILECOIN_NUMBER_PROPTYPE } from '../../customPropTypes'

type FilecoinDenomination = 'fil' | 'picofil' | 'attofil'

/**
 * Since Infinity and NaN don't exist as useful defaults,
 * we use this method to check if the prop has been passed
 **/
const isFilecoinNumber = value =>
  typeof value === 'object' &&
  'toFil' in value &&
  'toAttoFil' in value &&
  'toPicoFil' in value

/**
 * Get the string representation of the Filecoin number for the required denomination
 **/
const getValue = (value: FilecoinNumber, denom: FilecoinDenomination) => {
  switch (denom) {
    case 'fil':
      return value.toFil()
    case 'picofil':
      return value.toPicoFil()
    case 'attofil':
      return value.toAttoFil()
    default:
      return ''
  }
}

/**
 * Get the unit representation for the required denomination
 **/
const getUnit = (denom: FilecoinDenomination) => {
  switch (denom) {
    case 'fil':
      return 'FIL'
    case 'picofil':
      return 'pFIL'
    case 'attofil':
      return 'aFIL'
    default:
      return ''
  }
}

/**
 * FilecoinInput
 *
 * This input is based on the NumberInput, with the difference that
 * min, max and value are of type "FilecoinNumber" and not "number".
 *
 * Instead of triggering "onChange", "onFocus" and "onBlur" with NaN,
 * these methods will not be called when the input is empty or invalid.
 */
export const FilecoinInput = ({
  min,
  max,
  value,
  denom,
  onChange,
  onFocus,
  onBlur,
  setIsValid,
  ...baseProps
}: FilecoinInputProps) => {
  const [error, setError] = useState<string>('')
  const [showError, setShowError] = useState<boolean>(false)

  const onChangeBase = (newValue: string) => {
    try {
      onChange(newValue ? new FilecoinNumber(newValue, denom) : null)
    } catch (e) {
      onChange(null)
    }
  }

  const onFocusBase = () => {
    setShowError(false)
    onFocus()
  }

  const onBlurBase = () => {
    setShowError(true)
    onBlur()
  }

  useEffect(() => {
    setError('')
    setIsValid(true)
  }, [value])

  return (
    <BaseInput
      error={showError ? error : ''}
      type='number'
      unit={getUnit(denom)}
      value={isFilecoinNumber(value) ? getValue(value, denom) : ''}
      onChange={onChangeBase}
      onFocus={onFocusBase}
      onBlur={onBlurBase}
      {...baseProps}
    />
  )
}

/**
 * We strip certain properties from the standard
 * base input props because we want to override them:
 *
 * error: set by filecoin input validation
 * type: always "number" for filecoin input
 * unit: will be based on the FilecoinDenomination
 * value: needs to be of type "FilecoinNumber" / "FILECOIN_NUMBER_PROPTYPE"
 * onChange: needs to take "FilecoinNumber" type argument
 *
 * We add min, max, denom and setIsValid
 */

export type FilecoinInputProps = {
  min?: FilecoinNumber | null
  max?: FilecoinNumber | null
  value?: FilecoinNumber | null
  denom?: FilecoinDenomination
  onChange?: (value: FilecoinNumber | null) => void
  setIsValid?: (hasError: boolean) => void
} & Omit<BaseInputProps, 'error' | 'type' | 'unit' | 'value' | 'onChange'>

// "onChange" remains "PropTypes.func", so doesn't need an override
const { error, type, unit, value, ...filecoinProps } = BaseInputPropTypes

FilecoinInput.propTypes = {
  min: FILECOIN_NUMBER_PROPTYPE,
  max: FILECOIN_NUMBER_PROPTYPE,
  value: FILECOIN_NUMBER_PROPTYPE,
  denom: PropTypes.oneOf(['fil', 'picofil', 'attofil']),
  setIsValid: PropTypes.func,
  ...filecoinProps
}

/**
 * Provide defaults for props that are used in this input
 */

FilecoinInput.defaultProps = {
  min: null,
  max: null,
  value: null,
  denom: 'fil',
  onChange: () => {},
  onFocus: () => {},
  onBlur: () => {},
  setIsValid: () => {}
}
