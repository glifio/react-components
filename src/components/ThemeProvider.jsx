import PropTypes from 'prop-types'
import { ThemeProvider as StyledThemeProvider } from 'styled-components'
import { theme as defaultTheme } from './theme'

const ThemeProvider = ({ children, theme, ...rest }) => {
  return (
    <StyledThemeProvider theme={theme} {...rest}>
      {children}
    </StyledThemeProvider>
  )
}

ThemeProvider.propTypes = {
  theme: PropTypes.object,
  children: PropTypes.node
}

ThemeProvider.defaultProps = {
  theme: defaultTheme,
  children: <></>
}

ThemeProvider.displayName = 'ThemeProvider'

export default ThemeProvider
