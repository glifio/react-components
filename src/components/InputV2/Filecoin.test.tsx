import { cleanup, render, act, getByRole, RenderResult } from '@testing-library/react'
import { FilecoinNumber } from '@glif/filecoin-number'
import { FilecoinInput } from './Filecoin'
import ThemeProvider from '../ThemeProvider'
import theme from '../theme'

const labelText = 'Enter an amount in Filecoin'
const infoText = 'This is how much will be transferred'

describe('Filecoin input', () => {
  afterEach(cleanup)
  let setIsValid = jest.fn()

  beforeEach(() => {
    jest.clearAllMocks()
    setIsValid = jest.fn()
  })

  test('it renders fil correctly', async () => {
    let result: RenderResult | null = null
    await act(async () => {
      result = render(
        <ThemeProvider theme={theme}>
          <FilecoinInput
            label={labelText}
            info={infoText}
            denom='fil'
            value={new FilecoinNumber(100, 'fil')}
            setIsValid={setIsValid}
          />
        </ThemeProvider>
      )
    })
    expect(setIsValid).toHaveBeenCalledTimes(1)
    expect(setIsValid).toHaveBeenCalledWith(true)
    expect(result.container.firstChild).toMatchSnapshot()
  })

  test('it renders attofil correctly', async () => {
    let result: RenderResult | null = null
    await act(async () => {
      result = render(
        <ThemeProvider theme={theme}>
          <FilecoinInput
            label={labelText}
            info={infoText}
            denom='attofil'
            value={new FilecoinNumber(100, 'attofil')}
            setIsValid={setIsValid}
          />
        </ThemeProvider>
      )
    })
    expect(setIsValid).toHaveBeenCalledTimes(1)
    expect(setIsValid).toHaveBeenCalledWith(true)
    expect(result.container.firstChild).toMatchSnapshot()
  })

  test('it renders picofil correctly', async () => {
    let result: RenderResult | null = null
    await act(async () => {
      result = render(
        <ThemeProvider theme={theme}>
          <FilecoinInput
            label={labelText}
            info={infoText}
            denom='picofil'
            value={new FilecoinNumber(100, 'picofil')}
            setIsValid={setIsValid}
          />
        </ThemeProvider>
      )
    })
    expect(setIsValid).toHaveBeenCalledTimes(1)
    expect(setIsValid).toHaveBeenCalledWith(true)
    expect(result.container.firstChild).toMatchSnapshot()
  })

  test('it renders the value too high state correctly', async () => {
    let result: RenderResult | null = null
    await act(async () => {
      result = render(
        <ThemeProvider theme={theme}>
          <FilecoinInput
            label={labelText}
            info={infoText}
            denom='fil'
            max={new FilecoinNumber(10, 'fil')}
            value={new FilecoinNumber(100, 'fil')}
            setIsValid={setIsValid}
            autofocus={true}
          />
        </ThemeProvider>
      )
      // Make sure the error is shown
      getByRole(result.container, 'spinbutton').blur()
    })
    expect(setIsValid).toHaveBeenCalledTimes(1)
    expect(setIsValid).toHaveBeenCalledWith(false)
    expect(result.container.firstChild).toMatchSnapshot()
  })

  test('it renders the value too low state correctly', async () => {
    let result: RenderResult | null = null
    await act(async () => {
      result = render(
        <ThemeProvider theme={theme}>
          <FilecoinInput
            label={labelText}
            info={infoText}
            denom='fil'
            min={new FilecoinNumber(100, 'fil')}
            value={new FilecoinNumber(10, 'fil')}
            setIsValid={setIsValid}
            autofocus={true}
          />
        </ThemeProvider>
      )
      // Make sure the error is shown
      getByRole(result.container, 'spinbutton').blur()
    })
    expect(setIsValid).toHaveBeenCalledTimes(1)
    expect(setIsValid).toHaveBeenCalledWith(false)
    expect(result.container.firstChild).toMatchSnapshot()
  })

  test('it renders the disabled state correctly', async () => {
    let result: RenderResult | null = null
    await act(async () => {
      result = render(
        <ThemeProvider theme={theme}>
          <FilecoinInput
            label={labelText}
            info={infoText}
            disabled={true}
          />
        </ThemeProvider>
      )
      expect(result.container.firstChild).toMatchSnapshot()
    })
  })
})
