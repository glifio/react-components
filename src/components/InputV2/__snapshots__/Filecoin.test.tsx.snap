// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filecoin input it renders attofil correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: var(--space-m);
  text-align: left;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c0 > *:first-child {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 > *:last-child:not(.toggle-wrapper) {
  -webkit-flex: 0 0 72%;
  -ms-flex: 0 0 72%;
  flex: 0 0 72%;
}

.c0 span.info,
.c0 span.error {
  font-size: 0.875rem;
}

.c0 span.error {
  color: var(--red-medium);
}

.c1 .button-wrapper {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: var(--space-m);
}

.c1 .button-wrapper > *:first-child {
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
}

.c1 .button-wrapper > *:not(:first-child) {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-transition: -webkit-transform 0.1s ease-out;
  -webkit-transition: transform 0.1s ease-out;
  transition: transform 0.1s ease-out;
}

.c1 .button-wrapper > *:not(:first-child):hover:not(:active) {
  -webkit-transform: scale(1.2);
  -ms-transform: scale(1.2);
  transform: scale(1.2);
}

.c1 .button-wrapper > *:not(:first-child):active {
  -webkit-transition: none;
  transition: none;
}

.c1 .unit-wrapper {
  position: relative;
}

.c1 .unit-wrapper input {
  width: 100%;
}

.c1 .unit-wrapper .unit {
  position: absolute;
  top: 50%;
  right: 1em;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  color: var(--purple-medium);
}

<label
  class="c0 c1"
>
  <div>
    <span>
      Enter an amount in Filecoin
    </span>
    <span
      class="info"
    >
      This is how much will be transferred
    </span>
    
  </div>
  <div
    class="button-wrapper"
  >
    <div
      class="unit-wrapper"
    >
      <input
        autocomplete="on"
        class=""
        max=""
        min="0"
        name=""
        placeholder=""
        step="any"
        style="padding-right: 4em;"
        type="number"
        value="100"
      />
      <span
        class="unit"
      >
        aFIL
      </span>
    </div>
  </div>
</label>
`;

exports[`Filecoin input it renders fil correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: var(--space-m);
  text-align: left;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c0 > *:first-child {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 > *:last-child:not(.toggle-wrapper) {
  -webkit-flex: 0 0 72%;
  -ms-flex: 0 0 72%;
  flex: 0 0 72%;
}

.c0 span.info,
.c0 span.error {
  font-size: 0.875rem;
}

.c0 span.error {
  color: var(--red-medium);
}

.c1 .button-wrapper {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: var(--space-m);
}

.c1 .button-wrapper > *:first-child {
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
}

.c1 .button-wrapper > *:not(:first-child) {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-transition: -webkit-transform 0.1s ease-out;
  -webkit-transition: transform 0.1s ease-out;
  transition: transform 0.1s ease-out;
}

.c1 .button-wrapper > *:not(:first-child):hover:not(:active) {
  -webkit-transform: scale(1.2);
  -ms-transform: scale(1.2);
  transform: scale(1.2);
}

.c1 .button-wrapper > *:not(:first-child):active {
  -webkit-transition: none;
  transition: none;
}

.c1 .unit-wrapper {
  position: relative;
}

.c1 .unit-wrapper input {
  width: 100%;
}

.c1 .unit-wrapper .unit {
  position: absolute;
  top: 50%;
  right: 1em;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  color: var(--purple-medium);
}

<label
  class="c0 c1"
>
  <div>
    <span>
      Enter an amount in Filecoin
    </span>
    <span
      class="info"
    >
      This is how much will be transferred
    </span>
    
  </div>
  <div
    class="button-wrapper"
  >
    <div
      class="unit-wrapper"
    >
      <input
        autocomplete="on"
        class=""
        max=""
        min="0"
        name=""
        placeholder=""
        step="any"
        style="padding-right: 3.25em;"
        type="number"
        value="100"
      />
      <span
        class="unit"
      >
        FIL
      </span>
    </div>
  </div>
</label>
`;

exports[`Filecoin input it renders picofil correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: var(--space-m);
  text-align: left;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c0 > *:first-child {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 > *:last-child:not(.toggle-wrapper) {
  -webkit-flex: 0 0 72%;
  -ms-flex: 0 0 72%;
  flex: 0 0 72%;
}

.c0 span.info,
.c0 span.error {
  font-size: 0.875rem;
}

.c0 span.error {
  color: var(--red-medium);
}

.c1 .button-wrapper {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: var(--space-m);
}

.c1 .button-wrapper > *:first-child {
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
}

.c1 .button-wrapper > *:not(:first-child) {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-transition: -webkit-transform 0.1s ease-out;
  -webkit-transition: transform 0.1s ease-out;
  transition: transform 0.1s ease-out;
}

.c1 .button-wrapper > *:not(:first-child):hover:not(:active) {
  -webkit-transform: scale(1.2);
  -ms-transform: scale(1.2);
  transform: scale(1.2);
}

.c1 .button-wrapper > *:not(:first-child):active {
  -webkit-transition: none;
  transition: none;
}

.c1 .unit-wrapper {
  position: relative;
}

.c1 .unit-wrapper input {
  width: 100%;
}

.c1 .unit-wrapper .unit {
  position: absolute;
  top: 50%;
  right: 1em;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  color: var(--purple-medium);
}

<label
  class="c0 c1"
>
  <div>
    <span>
      Enter an amount in Filecoin
    </span>
    <span
      class="info"
    >
      This is how much will be transferred
    </span>
    
  </div>
  <div
    class="button-wrapper"
  >
    <div
      class="unit-wrapper"
    >
      <input
        autocomplete="on"
        class=""
        max=""
        min="0"
        name=""
        placeholder=""
        step="any"
        style="padding-right: 4em;"
        type="number"
        value="100"
      />
      <span
        class="unit"
      >
        pFIL
      </span>
    </div>
  </div>
</label>
`;

exports[`Filecoin input it renders the disabled state correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: var(--space-m);
  text-align: left;
  pointer-events: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c0 > *:first-child {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 > *:last-child:not(.toggle-wrapper) {
  -webkit-flex: 0 0 72%;
  -ms-flex: 0 0 72%;
  flex: 0 0 72%;
}

.c0 span.info,
.c0 span.error {
  font-size: 0.875rem;
}

.c0 span.error {
  color: var(--red-medium);
}

.c1 .button-wrapper {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: var(--space-m);
}

.c1 .button-wrapper > *:first-child {
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
}

.c1 .button-wrapper > *:not(:first-child) {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-transition: -webkit-transform 0.1s ease-out;
  -webkit-transition: transform 0.1s ease-out;
  transition: transform 0.1s ease-out;
}

.c1 .button-wrapper > *:not(:first-child):hover:not(:active) {
  -webkit-transform: scale(1.2);
  -ms-transform: scale(1.2);
  transform: scale(1.2);
}

.c1 .button-wrapper > *:not(:first-child):active {
  -webkit-transition: none;
  transition: none;
}

.c1 .unit-wrapper {
  position: relative;
}

.c1 .unit-wrapper input {
  width: 100%;
}

.c1 .unit-wrapper .unit {
  position: absolute;
  top: 50%;
  right: 1em;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  color: var(--purple-medium);
  color: var(--gray-dark) !important;
}

<label
  class="c0 c1"
  disabled=""
>
  <div>
    <span>
      Enter an amount in Filecoin
    </span>
    <span
      class="info"
    >
      This is how much will be transferred
    </span>
    
  </div>
  <div
    class="button-wrapper"
  >
    <div
      class="unit-wrapper"
    >
      <input
        autocomplete="on"
        class=""
        disabled=""
        max=""
        min="0"
        name=""
        placeholder=""
        step="any"
        style="padding-right: 3.25em;"
        type="number"
        value=""
      />
      <span
        class="unit"
      >
        FIL
      </span>
    </div>
  </div>
</label>
`;

exports[`Filecoin input it renders the value too high state correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: var(--space-m);
  text-align: left;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c0 > *:first-child {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 > *:last-child:not(.toggle-wrapper) {
  -webkit-flex: 0 0 72%;
  -ms-flex: 0 0 72%;
  flex: 0 0 72%;
}

.c0 span.info,
.c0 span.error {
  font-size: 0.875rem;
}

.c0 span.error {
  color: var(--red-medium);
}

.c1 .button-wrapper {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: var(--space-m);
}

.c1 .button-wrapper > *:first-child {
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
}

.c1 .button-wrapper > *:not(:first-child) {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-transition: -webkit-transform 0.1s ease-out;
  -webkit-transition: transform 0.1s ease-out;
  transition: transform 0.1s ease-out;
}

.c1 .button-wrapper > *:not(:first-child):hover:not(:active) {
  -webkit-transform: scale(1.2);
  -ms-transform: scale(1.2);
  transform: scale(1.2);
}

.c1 .button-wrapper > *:not(:first-child):active {
  -webkit-transition: none;
  transition: none;
}

.c1 .unit-wrapper {
  position: relative;
}

.c1 .unit-wrapper input {
  width: 100%;
}

.c1 .unit-wrapper .unit {
  position: absolute;
  top: 50%;
  right: 1em;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  color: var(--purple-medium);
  color: var(--red-dark) !important;
}

<label
  class="c0 c1"
>
  <div>
    <span>
      Enter an amount in Filecoin
    </span>
    <span
      class="info"
    >
      This is how much will be transferred
    </span>
    <span
      class="error"
    >
      Cannot be more than 10 FIL
    </span>
  </div>
  <div
    class="button-wrapper"
  >
    <div
      class="unit-wrapper"
    >
      <input
        autocomplete="on"
        class="error"
        max="10"
        min="0"
        name=""
        placeholder=""
        step="any"
        style="padding-right: 3.25em;"
        type="number"
        value="100"
      />
      <span
        class="unit"
      >
        FIL
      </span>
    </div>
  </div>
</label>
`;

exports[`Filecoin input it renders the value too low state correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: var(--space-m);
  text-align: left;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c0 > *:first-child {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 > *:last-child:not(.toggle-wrapper) {
  -webkit-flex: 0 0 72%;
  -ms-flex: 0 0 72%;
  flex: 0 0 72%;
}

.c0 span.info,
.c0 span.error {
  font-size: 0.875rem;
}

.c0 span.error {
  color: var(--red-medium);
}

.c1 .button-wrapper {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: var(--space-m);
}

.c1 .button-wrapper > *:first-child {
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
}

.c1 .button-wrapper > *:not(:first-child) {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-transition: -webkit-transform 0.1s ease-out;
  -webkit-transition: transform 0.1s ease-out;
  transition: transform 0.1s ease-out;
}

.c1 .button-wrapper > *:not(:first-child):hover:not(:active) {
  -webkit-transform: scale(1.2);
  -ms-transform: scale(1.2);
  transform: scale(1.2);
}

.c1 .button-wrapper > *:not(:first-child):active {
  -webkit-transition: none;
  transition: none;
}

.c1 .unit-wrapper {
  position: relative;
}

.c1 .unit-wrapper input {
  width: 100%;
}

.c1 .unit-wrapper .unit {
  position: absolute;
  top: 50%;
  right: 1em;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  color: var(--purple-medium);
  color: var(--red-dark) !important;
}

<label
  class="c0 c1"
>
  <div>
    <span>
      Enter an amount in Filecoin
    </span>
    <span
      class="info"
    >
      This is how much will be transferred
    </span>
    <span
      class="error"
    >
      Cannot be less than 100 FIL
    </span>
  </div>
  <div
    class="button-wrapper"
  >
    <div
      class="unit-wrapper"
    >
      <input
        autocomplete="on"
        class="error"
        max=""
        min="100"
        name=""
        placeholder=""
        step="any"
        style="padding-right: 3.25em;"
        type="number"
        value="10"
      />
      <span
        class="unit"
      >
        FIL
      </span>
    </div>
  </div>
</label>
`;
